### PR TITLE
Support themes

### PR DESCRIPTION
**Pull request by lowjoel**
Since there's now some work on the official Theme repository.

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -9,3 +9,17 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 # Rails.application.config.assets.precompile += %w( search.js )
+
+# For themes. See https://github.com/yoolk/themes_on_rails
+Rails.application.configure do
+  themes_path = "#{Rails.root}/app/themes/"
+  assets_paths = [
+    proc do |path, filename|
+      filename =~ /app\/themes/ && !['.js', '.css'].include?(File.extname(path))
```
Comment (lowjoel) [inline config/initializers/assets.rb]: @allenwq I think we should disable this? %w and %r are cryptic, if you ask me...

**Comment by weiqingtoh** [comment]
Quick question - I'm proposing to make some minor changes to the themes in `app/themes/default/views/layouts/default.html.slim`, and also clarify which elements will be handled by themes as well as the app views. Moving forward, for changes to themes, we will do PRs to the `coursemology/coursemology-theme` repo?

(Sorry I haven't pushed that up for discussion cause I'm facing some issues with a spec ><)

**Comment by lowjoel** [comment]
It's going to be layered. Default will only be used by the default tenant and tenants without a theme. Changing that is fine if you want it applied to all installs of Coursemology.

The themes that we define separately will not be affected, and templates there will override those in default. So you'd need to make changes in two different repositories if you are making a theme change.

I expect most themes not to override too many templates, probably only the main layout.

File: config/initializers/assets.rb (hunk 1 above)
Comment (allenwq) [inline config/initializers/assets.rb]: Yep, we have the convention not using %w, I think same for %r.

**Comment by allenwq** [comment]
@lowjoel Do we want to make each theme a gem?

**Comment by lowjoel** [comment]
Yes, they will have to be. We only will have one theme -- the official Coursemology.org theme.

Other people can set their own instances up and install their own.

File: config/initializers/assets.rb (hunk 1 above)
Comment (lowjoel) [inline config/initializers/assets.rb]: Maybe I'll add this to themes_on_rails as a Railtie later. Maybe you can PR the changing of Rubocop's rules?

Comment (allenwq) [inline config/initializers/assets.rb]: Sure

**Pull request merged**